src/middleware/FileRestrictionsMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys';
+import { UserProfileKeys } from 'ch-node-session-handler/lib/session/keys/UserProfileKeys';
+import { ISignInInfo, IUserProfile } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { NextFunction, Request, Response } from 'express';
+import { FORBIDDEN } from 'http-status-codes';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+import { loggerInstance } from './Logger';
+
+import { Appeal } from 'app/models/Appeal';
+import { AppealsPermissionKeys } from 'app/models/AppealPermissionKeys';
+import { APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { Attachment } from 'app/models/Attachment';
+
+const customErrorTemplate = 'error-custom';
+
+@provide(FileRestrictionsMiddleware)
+export class FileRestrictionsMiddleware extends BaseMiddleware {
+
+    public handler(req: Request, res: Response, next: NextFunction): void {
+
+        const session = req.session;
+
+        const adminPermissionFlag = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.AdminPermissions])
+            .extract();
+
+        const userProfile: IUserProfile = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.UserProfile])
+            .ifNothing(() => loggerInstance().error(`${FileRestrictionsMiddleware.name} - User profile was expected in session but none found`))
+            .unsafeCoerce();
+
+        const permissions = userProfile[UserProfileKeys.Permissions];
+
+        if (adminPermissionFlag === '1') {
+            if (permissions &&
+                permissions[AppealsPermissionKeys.download] &&
+                permissions[AppealsPermissionKeys.view]) {
+                return next();
+            }
+            return this.renderForbiddenError(res);
+
+        } else {
+
+            const appeal: Appeal = session.chain(_ => _.getExtraData())
+                .chainNullable(extraData => extraData[APPLICATION_DATA_KEY])
+                .chainNullable(applicationData => applicationData.appeal)
+                .ifNothing(() => loggerInstance().error(`${FileRestrictionsMiddleware.name} - Appeal was expected in session but none found`))
+                .unsafeCoerce();
+
+            const fileId = req.params.fileId;
+            const attachement = this.getAttachmentFrom(appeal, fileId);
```

Comment:
please rename this method to `getAttachment` and add the type

Reply:
Also, what's the reason for only just checking if there's an attachment on the appeal for the field id?

Reply:
Can this check be done at controller level?

Reply:
The purpose of the middleware is to check whether the user has the correct permissions to access the file.

Reply:
to check that the external user has permissions for the first case, we have to check that they are trying to access an appeal that belongs to that user and that the file belongs to that appeal

Reply:
I wonder if we could make this more readable by creating methods for each of the cases?